### PR TITLE
chore: bump loaders and optimize webpack configuration

### PR DESCRIPTION
**Pull request by AtofStryker**
- Closes N/A

### Additional details

As a follow up to #27438, this PR follows the [clean up configuration guide](https://webpack.js.org/migrate/5/#clean-up-configuration) set forth by webpack on the 4 -> 5 update and is the last step in the upgrade. This includes:

##### loaders
* `ts-loader` from `v8` to `v9` fbff60fb3ccfba4b74a8ddb81c7c1a41908ae8cc [release notes](https://github.com/TypeStrong/ts-loader/releases/tag/v9.0.0) (node version deprecation and min webpack 5 support)
* `sass-loader` from `v10` to `v13`1afd7d3958d369718898b74fd381ef61713d5f74 [release notes](https://github.com/webpack-contrib/sass-loader/releases)  (node version deprecation and min webpack 5 support). `tilda` imports are deprecated and aren't needed (same with `css-loader`) and have been removed [see v11 release notes](https://github.com/webpack-contrib/sass-loader/releases/tag/v11.0.0)
* `postcss-loader` from `v4` to `v7` ed6ed1b1e3f09c6e29b8848ba186f18d57a8ec56 [release notes](https://github.com/webpack-contrib/postcss-loader/releases) (node version deprecation and min webpack 5 support)
* `css-loader` from `v5` to `v6`3c6d1eedd3316a405e59dc0e13ba7c5322f9d74e [release notes](https://github.com/webpack-contrib/css-loader/releases/tag/v6.0.0)  (option changes that don't effect us, node version deprecation and min webpack 5 support). the `esModules` config is enabled by default, which we can opt to turn off if needed. `tilda` imports are deprecated now, which was addressed with the `sass-loader` commit.
* `babel-loader` from `v8` to `v9` 37840fd88fc32864521b1e31e9de104389abad0d [release notes](https://github.com/babel/babel-loader/releases/tag/v9.0.0)
    * to use `babel-loader` `v9`, babel `^7.12.0` must be used and greater than Node `v14` is needed.
    * This update to the loader includes and update to babel `latest@7`, which entails:
        *  `@babel/plugin-proposal-class-properties` is no longer a proposal and is now `@babel/plugin-transform-class-properties`
        *  `@babel/plugin-proposal-object-rest-spread` is no longer a proposal and is now `@babel/plugin-transform-object-rest-spread`
        * `@babel/plugin-transform-private-methods` and `@babel/plugin-transform-private-property-in-object` are added into `webpack.config.base.ts` to reduce warnings of `loose: true` for `@babel/plugin-proposal-class-properties` and `@babel/plugin-proposal-object-rest-spread` as `loose: false` is the default. If `loose: true` is enabled for `@babel/plugin-proposal-class-properties` and `@babel/plugin-proposal-object-rest-spread`, it must be enabled for `@babel/plugin-transform-private-methods` and `@babel/plugin-transform-private-property-in-object` as well.
        * Since some of our CT `webpack@4` tests had a dependency on `babel-loader@8`, I needed to pin it in the `package.json` of the system-tests to keep the dependency trees as expected 31e3a3e7e394794410a6d1642236fd1c5ec50246
* `coffee-loader`  from `v1` to `v4` and `coffeescript` from `v1` to `v2.6` 9a7f0bf08ec5fb816d0c8018772add76534646f2 [release notes](https://github.com/webpack-contrib/coffee-loader/releases). `coffeescript@2.6.1` has issues with packherd and didn't seem worth investigating at this time
* Removal of `file-loader` as it deprecated will no longer be supported 4a8ecfef6b105950ae0b3ae586e141e529d75b68. This means a few things:
    * inside the `webpack.config.base.ts`, we now use the preferred `asset/resource` type for:
        * fonts:  `eot`, `ttf`,`woff`, and `woff2`
        * images: `png` and `gif` (the rule is configured but there are no image assets in the build), hence the removal `pngRule` inside `@packages/runner`

        The options are slightly different for `asset/resource` -> `filename` then they are for `file-loader` -> `options.name`, but the files resolve to the same spot. This can be verified by running `yarn build` inside `@packages/runner`
    * `svg`s are **NOT** handled by `asset/resource` loading (as none exist to my knowledge). What we were previously doing is leverage [react-svg-loader](https://github.com/boopathi/react-svg-loader/tree/master) (which is no longer maintained) inside `@packages/runner` src files to tell webpack how to process the file since we didn't have the loader declared in config. To modernize the loader config, 4a8ecfef6b105950ae0b3ae586e141e529d75b68 also:
        * removes `react-svg-loader` including direct references to loader importing inside `@packages/reporter` src code which further decouples the src from the bundling (most agree this is a good thing)
        *  leverage [`@svgr/webpack`](https://www.npmjs.com/package/@svgr/webpack) over `react-svg-loader` for `jsx`/`tsx` files inside `@packages/reporter`. Since we no longer have explicit loader references inside the source, our reporter E2E tests need to know how to process `svg`s inside `tsx`/`jsx` files under test. For this, we pass the `@packages/reporter` webpack configuration into the `@cypress/webpack-preprocessor` inside the `file:preprocessor` plugin event inside `cypress.config.ts`
 
##### Plugins
* `mini-css-extract-plugin` from `v1` to `v2` 811cdca84515ac11aa0b71f3105057177ea5a42c [release notes](https://github.com/webpack-contrib/mini-css-extract-plugin/releases/tag/v2.0.0) (option changes that do not effect our config, node version deprecation and min webpack 5 support)
* `html-webpack-plugin` from `v4` to `v5`8ba2524c14299d3511db74cecfe820a8b2bc6b1e [release notes](https://github.com/jantimon/html-webpack-plugin/blob/main/CHANGELOG.md#500-2021-02-03)  (option changes that do not effect our config, node version deprecation and min webpack 5 support)
* `copy-webpack-plugin` from `v6` to `v11` 379dc9462681d5046d5b1e3499da098fd96ea3bc [release notes](https://github.com/webpack-contrib/copy-webpack-plugin/releases/tag/v7.0.0) (option changes that don't effect us, node version deprecation and min webpack 5 support)

#### Making small optimizations  and fixing build warnings

* Warning: Remove `Url` import from validation code as it should be available as global built in d85c75cd72df319853b4e3a18b41fddf2ef60f85
* Optimization: Switches over config using `hash` to use `contenthash` as recommended by [clean up configuration guide](https://webpack.js.org/migrate/5/#clean-up-configuration) b668c30dc9c287b5320b310bedaab1d3c54941b2
* Optimization: Removes the use of `moduleIds` and `chunkIds` from `webpack.config.base.ts` as the defaults are likely better, which is also recommended in [clean up configuration guide](https://webpack.js.org/migrate/5/#clean-up-configuration) 3f8a2e54cfc8fa1b481e3470f74f7cf6c3fe4ae2

#### Misc Changes
* Update the Node versions under test as a lot of the loaders now no longer support Node 12 or Node 14. [We no longer support Node 12 or 14](https://docs.cypress.io/guides/getting-started/installing-cypress#Nodejs), but our binary tests were not up to date. The loader changes forces us to update them in ca0ebc303e1f45118fed2288d3a8583fe6fe8d1c

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: system-tests/test-binary/ci_environments_spec.ts [hunk 1]
```diff
@@ -16,26 +16,27 @@ function smokeTestDockerImage (title: string, dockerImage: string, expectedExitC
...
+    'node:16', 1,
     async (exec) => {
       const { stdout } = await exec()
 
       expect(stdout).to.include('Your system is missing the dependency: Xvfb')
     },
   )
 
+  // TODO: Where is this image located? Needs to be bumped to Node 16.16.0 or later
   smokeTestDockerImage(
```
Comment (AtofStryker) [inline system-tests/test-binary/ci_environments_spec.ts:28]: anyone have any ideas? this doesn't live in the docker images repo AFAIK

File: system-tests/test/form_submissions_spec.js [hunk 2]
```diff
@@ -149,6 +151,8 @@ describe('e2e forms', () => {
     })
 
     systemTests.it('passes with http on localhost', {
+      // FIXME: webkit is 404ing on redirects after the form submit on the default clientRoute /__/
+      browser: '!webkit',
```
Comment (AtofStryker) [inline system-tests/test/form_submissions_spec.js:155]: I can't get these to pass on `develop` locally as well with webkit (pre webpack 5 updates)

**Review by chrisbreiding — APPROVED** [review]

**Comment by AtofStryker** [comment]
green CI build https://app.circleci.com/pipelines/github/cypress-io/cypress?branch=chore%2Fbump_loaders_and_optimize_webpack

**Review by nagash77 — APPROVED** [review]

**Comment by AtofStryker** [comment]
changes are rebased off of `develop` to get up to date and resolve conflicts. Commit references in readme are still accurate. Once linux job is green this is good to merge

**Review by cacieprins — APPROVED** [review]

**Pull request merged**